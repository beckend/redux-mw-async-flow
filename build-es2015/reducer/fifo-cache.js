"use strict";
/* tslint:disable: variable-name */
/* tslint:disable: no-reserved-keywords */
/**
 * Need a function to get from the Map which returns an object key valued
 * Haxx type the module since none exists
 */
const { FifoXCache: FifoXCacheOriginal } = require('x-cache');
exports.FifoXCacheOriginalClass = FifoXCacheOriginal;
class FifoXCache extends exports.FifoXCacheOriginalClass {
    constructor(...args) {
        super(...args);
    }
    getAll() {
        let entry = this.tail;
        const returned = {};
        while (entry) {
            returned[entry.key] = entry.value;
            entry = entry.older;
        }
        return returned;
    }
}
exports.FifoXCache = FifoXCache;
//# sourceMappingURL=fifo-cache.js.map