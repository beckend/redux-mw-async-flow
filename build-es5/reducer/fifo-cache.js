"use strict";
var tslib_1 = require("tslib");
/* tslint:disable: variable-name */
/* tslint:disable: no-reserved-keywords */
/**
 * Need a function to get from the Map which returns an object key valued
 * Haxx type the module since none exists
 */
var FifoXCacheOriginal = require('x-cache').FifoXCache;
exports.FifoXCacheOriginalClass = FifoXCacheOriginal;
var FifoXCache = (function (_super) {
    tslib_1.__extends(FifoXCache, _super);
    function FifoXCache() {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        return _super.apply(this, args) || this;
    }
    FifoXCache.prototype.getAll = function () {
        var entry = this.tail;
        var returned = {};
        while (entry) {
            returned[entry.key] = entry.value;
            entry = entry.older;
        }
        return returned;
    };
    return FifoXCache;
}(exports.FifoXCacheOriginalClass));
exports.FifoXCache = FifoXCache;
//# sourceMappingURL=fifo-cache.js.map