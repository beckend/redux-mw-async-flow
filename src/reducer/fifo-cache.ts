/* tslint:disable: variable-name */
/* tslint:disable: no-reserved-keywords */
/**
 * Need a function to get from the Map which returns an object key valued
 * Haxx type the module since none exists
 */
const { FifoXCache: FifoXCacheOriginal } = require('x-cache');

interface IFifoXCacheConstructor {
  new (...args: any[]): IFifoXCacheClass;
}
interface IFifoXCacheClass {
  readonly tail: any;
  readonly put: any;
  readonly set: any;
  readonly get: any;
  readonly getArray: any;
}
const FifoXCacheOriginalClass: IFifoXCacheConstructor = FifoXCacheOriginal;

class FifoXCache extends FifoXCacheOriginalClass {

  constructor(...args: any[]) {
    super(...args);
  }

  public getAll<TKeyPayload>() {
    let entry = this.tail;
    const returned: {
      [keyName: string]: TKeyPayload;
    } = {};
    while (entry) {
      returned[entry.key] = entry.value;
      entry = entry.older;
    }
    return returned;
  }

}

export { FifoXCache };
