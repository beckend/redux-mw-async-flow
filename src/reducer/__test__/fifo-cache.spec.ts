/* tslint:disable: chai-vague-errors */
/* tslint:disable: no-backbone-get-set-outside-model */
import { FifoXCache } from '../fifo-cache';

describe('FifoXCache', () => {
  it('extended class works as original', () => {
    const cache = new FifoXCache(2);
    cache.put('elwin', 1215);
    cache.put('leon', 1233);
    cache.put('tom', 1234);
    expect(cache.get('elwin')).toBeFalsy();
    expect(cache.get('leon')).toBeTruthy();
  });

  it('getAll added function works, returns unique keys', () => {
    const cache = new FifoXCache();
    cache.put('elwin', 1215);
    cache.put('leon', 1233);
    cache.put('tom', 1234);
    expect((cache.getAll()))
      .toMatchSnapshot();
  });
});
