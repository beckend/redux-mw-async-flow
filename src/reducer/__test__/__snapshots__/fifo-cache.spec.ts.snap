exports[`FifoXCache getAll added function works, returns unique keys 1`] = `
Object {
  "elwin": 1215,
  "leon": 1233,
  "tom": 1234,
}
`;
